lib/network_error.js

```diff
@@ -0,0 +1,44 @@
+"use strict";
```

Comment:
Is there a reason to have this live in its own file? Afaict, it should just live inside `error` and probably actually subclass `MongoError` itself (this is a specialization of that type)